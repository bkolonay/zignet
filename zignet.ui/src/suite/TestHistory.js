import React, { Component } from 'react';
import moment from 'moment';
import HistoryBarProvider from '../common/HistoryBarProvider'
import './testHistory.css';

class TestHistory extends Component {
  constructor() {
    super();
    this.state = {
      failureDivs: []
    };
    this.historyBarProvider = new HistoryBarProvider(); 
  }

  _getFailureDivs(testFailureDurations) {
    let historyBarWidth = this.historyBarDiv.offsetWidth
    let now = moment();
    var failureDivs = [];
    for (var i = 0; i < testFailureDurations.length; i++) {
      let failureDivAttributes = this.historyBarProvider.getErrorDivAttributes(historyBarWidth, now, testFailureDurations[i]);
      failureDivs.push(<div key={i} style={failureDivAttributes}/>);
    }
    return failureDivs;
  }

  componentDidMount() {
    this.setState({
      failureDivs: this._getFailureDivs(this.props.testFailureDurations)
    });
  }

  componentWillReceiveProps(props) {
    this.setState({
      failureDivs: this._getFailureDivs(props.testFailureDurations)
    });
  }

  render() {
    return (
      <div className="testHistoryBar"
           ref={(div) => this.historyBarDiv = div}>
      {this.state.failureDivs.map((failureDiv) => failureDiv)}      
      </div>
    );
  }
}

export default TestHistory;