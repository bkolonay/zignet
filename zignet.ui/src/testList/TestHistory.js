import React, { Component } from 'react';
import moment from 'moment';
import { getErrorDivAttributes, getDivTitle } from '../common/HistoryBarProvider'
import './css/testHistory.css';

class TestHistory extends Component {
  constructor() {
    super();
    this.state = {
      failureDivs: []
    };
  }

  _getFailureDivs(testFailureDurations) {
    let historyBarWidth = this.historyBarDiv.offsetWidth
    let now = moment();
    var failureDivs = [];
    for (var i = 0; i < testFailureDurations.length; i++) {
      let failureDivAttributes = getErrorDivAttributes(historyBarWidth, now, testFailureDurations[i]);
      failureDivs.push(<div key={i} style={failureDivAttributes} title={getDivTitle(testFailureDurations[i])}/>);
    }
    return failureDivs;
  }

  componentDidMount() {
    this.setState({
      failureDivs: this._getFailureDivs(this.props.testFailureDurations)
    });
  }

  componentWillReceiveProps(props) {
    this.setState({
      failureDivs: this._getFailureDivs(props.testFailureDurations)
    });
  }

  render() {
    return (
      <div className="testHistoryBar" ref={(div) => this.historyBarDiv = div}>
        {this.state.failureDivs.map((failureDiv) => failureDiv)}      
      </div>
    );
  }
}

export default TestHistory;