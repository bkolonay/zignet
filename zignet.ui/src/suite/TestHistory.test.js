import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TestHistory from './TestHistory';
jest.mock('react-router-dom');

// taken from: https://reactjs.org/blog/2016/11/16/react-v15.4.0.html#mocking-refs-for-snapshot-testing
function createNodeMock(element) {
  return {
    offsetWidth() {209}
  };
}

it('renders with no data', () => {

  const testFailureDurations = [];

  const div = document.createElement('div');
  ReactDOM.render(
    <TestHistory testFailureDurations={testFailureDurations}/>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders with single failure duration', () => {

  const testFailureDurations = [{
    FailureStart: '2018-05-10T01:00:00-07:00',
    FailureEnd: '2018-05-10T02:00:00-07:00'
  }];

  const div = document.createElement('div');
  ReactDOM.render(
    <TestHistory testFailureDurations={testFailureDurations}/>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders with multiple failure durations', () => {

  const testFailureDurations = [{
    FailureStart: '2018-05-10T01:00:00-07:00',
    FailureEnd: '2018-05-10T02:00:00-07:00'
  },
  {
    FailureStart: '2018-05-10T05:00:00-07:00',
    FailureEnd: '2018-05-10T07:00:00-07:00'
  }];

  const div = document.createElement('div');
  ReactDOM.render(
    <TestHistory testFailureDurations={testFailureDurations}/>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('matches the previous snapshot with no data', () => {

  const testFailureDurations = [];
  const options = {createNodeMock};

  const tree = renderer
    .create(<TestHistory testFailureDurations={testFailureDurations}/>, options)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('matches the previous snapshot with a single failure duration', () => {

  const testFailureDurations = [{
    FailureStart: '2018-05-15T01:00:00-07:00',
    FailureEnd: '2018-05-15T02:00:00-07:00'
  }];
  const options = {createNodeMock};

  const tree = renderer
    .create(<TestHistory testFailureDurations={testFailureDurations}/>, options)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('matches the previous snapshot with a multiple failure durations', () => {

  const testFailureDurations = [{
    FailureStart: '2018-05-15T01:00:00-07:00',
    FailureEnd: '2018-05-15T02:00:00-07:00'
  },
  {
    FailureStart: '2018-05-15T05:00:00-07:00',
    FailureEnd: '2018-05-15T07:00:00-07:00'
  }];
  const options = {createNodeMock};

  const tree = renderer
    .create(<TestHistory testFailureDurations={testFailureDurations}/>, options)
    .toJSON();
  expect(tree).toMatchSnapshot();
});