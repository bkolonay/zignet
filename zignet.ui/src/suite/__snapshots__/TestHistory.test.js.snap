// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the previous snapshot with a multiple failure durations 1`] = `
<div
  className="testHistoryBar"
>
  <div
    style={
      Object {
        "left": NaN,
        "width": NaN,
      }
    }
  />
  <div
    style={
      Object {
        "left": NaN,
        "width": NaN,
      }
    }
  />
</div>
`;

exports[`matches the previous snapshot with a single failure duration 1`] = `
<div
  className="testHistoryBar"
>
  <div
    style={
      Object {
        "left": NaN,
        "width": NaN,
      }
    }
  />
</div>
`;

exports[`matches the previous snapshot with no data 1`] = `
<div
  className="testHistoryBar"
/>
`;
